phoenix-core/src/main/java/org/apache/phoenix/iterate/OrderedResultIterator.java

```diff
@@ -245,23 +256,33 @@ public int compare(ImmutableBytesWritable o1, ImmutableBytesWritable o2) {
     
     @Override
     public Tuple next() throws SQLException {
-        return getResultIterator().next();
+        getResultIterator();
+        if (!resultIteratorReady) {
+            return dummyTuple;
+        }
+        return resultIterator.next();
     }
     
     private PeekingResultIterator getResultIterator() throws SQLException {
-        if (resultIterator != null) {
+        if (resultIteratorReady) {
```

Comment:
what does Ready mean in this context?

Reply:
The results have been ordered yet. When the results are ordered then the result iterator will be ready to iterate over them. I will add a comment for that.